lib/ruby_lsp/executor.rb

```diff
@@ -12,6 +12,7 @@ def initialize(store, message_queue)
       # store
       @store = store
       @message_queue = message_queue
+      @store.formatter = detected_formatter
```

Comment:
When formatter is `auto`, this is going to perform the detection on every single request - including when it's not formatting.

We need to move this to the initialize request, so that we only set it once. [Here](https://github.com/Shopify/ruby-lsp/blob/2b608f98501dd65f6561c33885ed172bbc407ac0/lib/ruby_lsp/executor.rb#L385) where we set the formatter, you can just invoke the logic and set the right one there.

Then only the formatting request will read `@store.formatter`.